libmicrohttpd: update to 1.0.1

MingcongBai opened the pull request:
Topic Description
-----------------

- libmicrohttpd: update to 1.0.1
    Co-authored-by: 白铭骢 (Mingcong Bai) (@MingcongBai) <jeffbai@aosc.io>

Package(s) Affected
-------------------

- libmicrohttpd: 1.0.1

Security Update?
----------------

No

Build Order
-----------

```
#buildit libmicrohttpd
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

aosc-buildit-bot committed:
libmicrohttpd: update to 1.0.1

Co-authored-by: 白铭骢 (Mingcong Bai) (@MingcongBai) <jeffbai@aosc.io>

OriginCode reviewed (changes requested):
The `beyond` and `prepare` scripts need a overhaul.

CAB233 committed:
libmicrohttpd: adjust the build to meet package styling

- Drop unnecessary prepare and beyond
- Adjust PKGDEP
- Improve AUTOTOOLS_AFTER
